tests/components/FilterSearch.test.tsx

```diff
@@ -0,0 +1,259 @@
+    render(<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+
+    userEvent.type(searchBarElement, '{backspace}');
+    expect(searchBarElement).toHaveValue('');
+
+    expect(mockedExecuteFilterSearch).toHaveBeenCalledTimes(0);
+  });
+
+  it('shows autocomplete suggestions when a character is typed and suggestions are returned by useSynchronizedSearch', () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+    expect(mockedExecuteFilterSearch).toBeCalledWith('n');
+
+    const firstAutocompleteSection= screen.getByText('First name');
+    expect(firstAutocompleteSection).toBeDefined();
+
+    const firstAutocompleteSuggestion = screen.getByText('first name 1');
+    expect(firstAutocompleteSuggestion).toBeDefined();
+
+    const secondAutocompleteSuggestion = screen.getByText('first name 2');
+    expect(secondAutocompleteSuggestion).toBeDefined();
+
+    const secondAutocompleteSection= screen.getByText('Last name');
+    expect(secondAutocompleteSection).toBeDefined();
+
+    const thirdAutocompleteSuggestion = screen.getByText('last name 1');
+    expect(thirdAutocompleteSuggestion).toBeDefined();
+  });
+
+  it('populates the search bar with selected suggestions when users navigate to populated autocomplete suggestions using arrow keys', () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    const firstAutocompleteSuggestion = screen.getByText('first name 1');
+    expect(firstAutocompleteSuggestion).toBeDefined();
+
+    const secondAutocompleteSuggestion = screen.getByText('first name 2');
+    expect(secondAutocompleteSuggestion).toBeDefined();
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 1');
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 2');
+  });
+
+  it('calls executeSearch when an autocomplete suggestion is selected using the arrow keys and "Enter"', () => {
+    render (<FilterSearch searchFields={searchFieldsProp}/>);
+
+    const searchBarElement = screen.getByRole('textbox');
+    userEvent.type(searchBarElement, 'n');
+
+    const firstAutocompleteSuggestion = screen.getByText('first name 1');
+    expect(firstAutocompleteSuggestion).toBeDefined();
+
+    userEvent.type(searchBarElement, '{arrowdown}');
+    expect(searchBarElement).toHaveValue('first name 1');
+    userEvent.type(searchBarElement, '{enter}');
+    expect(setFilterOption).toBeCalled();
+    expect(setOffset).toBeCalled();
+    expect(searchOperations.executeSearch).toBeCalled();
+  });
+
+  it('does not trigger executeSearch when "Enter" is pressed if no autocomplete suggestion is selected with the arrow keys first', () => {
```

Comment:
I think we can remove the "with the arrow keys first", since "selected" already kinda means that. We generally try to keep lines somewhere less than ~100 characters, this line is 139 characters!

Reply:
suuuper long tests are a sign that either 1. the name could be shortened and/or 2. the test is difficult to describe and understand and might be able to be broken up

Reply:
ahh that is good to know.  I'll keep that in mind for the future and get on all the other long lines in this file

Reply:
shortened all titles to under ~100 chars (maybe slightly over)

Reply:
Could this test just be `"does not trigger search when "enter" is pressed if no option is selected"`?

Reply:
We generally want to avoid specific implementation details like the "executeSearch" function unless they're part of the component's agreed upon interface (in other words the component's contract)

Reply:
got it, that makes sense.  I changed the names.